main/acle.md

```diff
@@ -4942,6 +4987,8 @@ following is true:
 
 *   F [uses](#uses-state) `"zt0"`
 
+*   F implements an [agnostic ZA](#agnostic-za) interface.
```

Comment:
Looks good, but this should be reflected in the table at the end of the section as well.

I think we also need to decide what to do about:

```
*   S is not an input to the asm and S is not an output from the asm.
    However, the contents of S after executing the asm's instructions might
    be different from the contents of S before executing the instructions.
    This is indicated by adding S's string to the asm's clobber list.

    If an asm takes this option for state that is controlled by PSTATE.ZA,
    the asm itself is responsible for handling the [[AAPCS64]](#AAPCS64)
    lazy save scheme.
```

The current rule implicity assumes that ZA can only be dormant or off.  If ZA is active, then is the asm required to preserve ZA despite the clobber?  Does the compiler do this?  (In which case, does it also handle the lazy-save scheme, meaning that the asm doesn't need to?)  Or is this option not valid in agnostic-ZA functions?

Reply:
My thinking on this was that for `__arm_agnostic("sme_za_state")` functions, if the asm clobbers any of the state enabled by PSTATE.ZA, the user is responsible for letting the function preserve this state using the provided routines such that on return from the function the ABI is honoured.